### DISCUSSION TITLE
Enabling Rond on CRUD Service

### DISCUSSION THREAD

**Discussion by rafpor75** [mia-platform/community#223] [Q&A] [answered]
### Which section?

Console

### Which is the version?

10.6.0

### Description

I am attempting to enable Rond on the CRUD service, following the instructions provided in the tutorial found at https://docs.mia-platform.eu/docs/tutorial/authorization/authorization_tutorial. 

### Actual Outcome

However, after enabling Rond, I encountered a deployment issue with the CRUD service. In the RBAC service log, I received the following error message: {"error":{"message":"missing environment variables one of TARGET_SERVICE_OAS_PATH or API_PERMISSIONS_FILE_PATH is required"},"level":50,"msg":"failed to load oas","oasApiPath":"","oasFilePath":"","time":1680169228732}. Additionally, the API Portal responded with a 500 error status.

**Comment by fredmaggiowski** [comment, marked as answer]
Hi @rafpor75 sorry for the delay, the sidecar is crashing because it requires either the OAS Endpoint from the service (which the console configures using the microservice documentation path, have you set it for the CRUD Service?) or the API Permissions File that the console creates when there is at least one manual route configured (from the Manual Routes tab in the Authorization section).

This is an odd scenario where the Console does not have any information about how you wish to configure the [rönd](//github.com/rond-authz/rond) sidecar and so it won't provide the required configurations for it to work.

@albertotessarotto FYI let's think of something to provide the user some insight of this configuration issue.

In the meantime @rafpor75 you can either configure the OAS Documentation Path in the CRUD Service configuration, set up your first manual route to protect one of your collections or disable the sidecar injection since, without the oas definition or policies you are not actually using it :)

**Reply by albertotessarotto** [reply]
If a service with rönd enabled has configured neither the field API documentation path, nor a manual route, in the overview table, on that service we can somehow pointing out that even if it is enabled, you are not really using it.
